src/lib.rs

```diff
@@ -225,15 +286,35 @@ fn enforce_container_security_policies(
     container: &mut apicore::Container,
     validation_request: &ValidationRequest<Settings>,
 ) -> Result<bool> {
+    let mut container_image_config = None;
+    if validation_request
+        .settings
+        .validate_container_image_configuration
+    {
+        let container_image = container.image.as_ref().unwrap();
+        let response = get_manifest_and_config(container_image)?;
+        if *response.config.os() != oci_spec::image::Os::Windows {
+            container_image_config = Some(response.config);
+        } else {
+            warn!(LOG_DRAIN, "Windows containers are not supported by the policy. Skipping container image configuration user validation."; "image" => &container_image);
+        }
+    }
+
     let mut mutated: bool = false;
-    let mutate_request =
-        enforce_run_as_user_rule(container.security_context.clone(), validation_request)?;
+    let mutate_request = enforce_run_as_user_rule(
+        container.security_context.clone(),
+        validation_request,
+        container_image_config.clone(), //TODO can I use reference here?
```

Comment:
I don't think you can, as once we return `mutated` it would be out of scope and the rust borrow checker would complain.